tests/functional/aws-node-sdk/lib/utility/makeRequest.js

```diff
@@ -0,0 +1,87 @@
+import http from 'http';
+import https from 'https';
+import querystring from 'querystring';
+
+import conf from '../../../../../lib/Config';
+
+const transport = conf.https ? https : http;
+const ipAddress = process.env.IP ? process.env.IP : '127.0.0.1';
+
+function _parseError(responseBody) {
+    if (responseBody.indexOf('<Error>') > -1) {
+        const error = {};
+        const codeStartIndex = responseBody.indexOf('<Code>') + 6;
+        const codeEndIndex = responseBody.indexOf('</Code>');
+        error.code = responseBody.slice(codeStartIndex, codeEndIndex);
+        const msgStartIndex = responseBody.indexOf('<Message>') + 9;
+        const msgEndIndex = responseBody.indexOf('</Message>');
+        error.message = responseBody.slice(msgStartIndex, msgEndIndex);
+        return error;
+    }
+    return null;
+}
+
+/** makeRequest - utility function to generate a request
+ * @param {object} params - params for making request
+ * @param {string} params.hostname - request hostname
+ * @param {number} params.port - request port
+ * @param {string} params.method - request method
+ * @param {object} params.queryObj - query fields and their string values
+ * @param {object} params.headers - headers and their string values
+ * @param {string} params.bucket - bucket name
+ * @param {string} params.objectKey - object key name
+ * @param {function} callback - with error and response parameters
+ * @return {undefined} - and call callback
+ */
+export default function makeRequest(params, callback) {
+    const { hostname, port, method, queryObj, headers, bucket, objectKey }
+        = params;
+    const options = {
+        hostname: hostname || ipAddress, // default to where S3 Server is hosted
+        port: port || 8000, // default to port 8000
```

Comment:
Ideally we should not write any fallbacks in tests. It would add some confusion.